### PR TITLE
Fix 27779 - JSP Stackoverflow

### LINKED ISSUE #27779: StackOverFlow in JSP Caused by Recurisve JspContextWrapper#include call
**Describe the bug**  
```
SRVE0777E: Exception thrown by application class 'org.apache.jasper.runtime.PageContextImpl.handlePageException:730'
com.ibm.websphere.servlet.error.ServletErrorReport: java.lang.StackOverflowError
	at org.apache.jasper.runtime.PageContextImpl.handlePageException(PageContextImpl.java:730)
        ...
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.lang.Thread.run(Thread.java:825)
Caused by: java.lang.StackOverflowError
	at org.apache.jasper.runtime.JspContextWrapper.include(JspContextWrapper.java:327)
	at org.apache.jasper.runtime.JspContextWrapper.include(JspContextWrapper.java:327)
	at org.apache.jasper.runtime.JspContextWrapper.include(JspContextWrapper.java:327)
```

**Steps to Reproduce**  
A JSP tag calls the following method:
[PageContext#include(String relativeUrlPath, boolean flush)](https://javaee.github.io/javaee-spec/javadocs/javax/servlet/jsp/PageContext.html#include-java.lang.String-boolean-)

**Expected behavior**  
A StackOverflowError should not occur. 

**Diagnostic information:**  
 - OpenLiberty Version: All 
 - Affected feature(s) jsp-2.3, pages-3.0, pages-3.1
 - Java Version: N/A
 - server.xml configuration: N/A
 
**Additional context**  
The code should the wrapped page context object instead. 
Also see the original BZ Issue:`https://bz.apache.org/bugzilla/show_bug.cgi?id=45666` (over 15 years old!)

### PR DESCRIPTION
**Pull request by volosied**
fixes #27779

### CODE REVIEW

**Comment by volosied** [comment]
#build

**Comment by dmuelle** [comment]
This PR comes up in our queue but I dont see any files that need ID review so adding the `ID review not needed` label. If that's not correct let me know.

File: dev/io.openliberty.pages.3.1.internal.factories/src/org/apache/jasper/runtime/JspContextWrapper.java [hunk 1]
```diff
@@ -327,7 +327,7 @@ public void include(String relativeUrlPath)
 
     public void include(String relativeUrlPath, boolean flush)
 	    throws ServletException, IOException {
-	include(relativeUrlPath, false); // XXX
+	invokingJspCtxt.include(relativeUrlPath, false); // XXX
```
Comment (epj) [inline dev/io.openliberty.pages.3.1.internal.factories/src/org/apache/jasper/runtime/JspContextWrapper.java]: Still have a XXX left.

**Comment by volosied** [comment]
#build

**Review by epj — APPROVED** [review]
ok

**Comment by volosied** [comment]
#libby

File: dev/com.ibm.ws.jsp.2.3_fat/test-applications/OLGH27779.war/resources/WEB-INF/web.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    
+    SPDX-License-Identifier: EPL-2.0
+ -->
+<web-app
+    version="3.1"
+    xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd">
+
+  <display-name>OLGH26891</display-name>
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsp.2.3_fat/test-applications/OLGH27779.war/resources/WEB-INF/web.xml:17]: @volosied the display name here is wrong and I'm assuming you wanted a 2024 copyright?

Comment (pnicolucci) [reply dev/com.ibm.ws.jsp.2.3_fat/test-applications/OLGH27779.war/resources/WEB-INF/web.xml:17]: https://github.com/OpenLiberty/open-liberty/pull/27955

**Pull request merged**